### PR TITLE
Align with Models and Connectors pattern 

### PR DESCRIPTION
**Pull request by AndrewOlenik**
Provide implementation of GraphQL Models and Connectors architecture (#26)

### CODE REVIEW

File: api/src/main/java/lv/ctco/tpl/bff/connectors/icndb/ICNDB.java [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
 import feign.Param;
 import feign.RequestLine;
 import lv.ctco.tpl.bff.graphql.types.JokeCategory;
-import lv.ctco.tpl.bff.integration.icndb.jokes.JokeResponseModel;
 
 public interface ICNDB {
 
     @RequestLine("GET /jokes/random")
-    JokeResponseModel getRandomJoke();
+    ICNDBJokeEnvelop getRandomJoke();
```
Comment (trioletas) [inline api/src/main/java/lv/ctco/tpl/bff/connectors/icndb/ICNDB.java]: typo `ICNDBJokeEnvelop` -> `ICNDBJokeEnvelope`

File: api/src/main/java/lv/ctco/tpl/bff/graphql/models/JokeModels.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import lv.ctco.tpl.bff.graphql.types.Joke;
+import lv.ctco.tpl.bff.graphql.types.JokeCategory;
+import lv.ctco.tpl.bff.graphql.types.JokeCategoryInput;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.stream.Collectors;
+
+@Component
+public class JokeModels {
```
Comment (trioletas) [inline api/src/main/java/lv/ctco/tpl/bff/graphql/models/JokeModels.java]: this should be singular: `JokeModel`, as models are mapped 1:1 to types. A single type will always have a single model.

File: api/src/main/java/lv/ctco/tpl/bff/connectors/icndb/ICNDB.java (hunk 1 above)
Comment (AndrewOlenik) [reply api/src/main/java/lv/ctco/tpl/bff/connectors/icndb/ICNDB.java]: thx, fixed.

File: api/src/main/java/lv/ctco/tpl/bff/graphql/models/JokeModels.java (hunk 2 above)
Comment (AndrewOlenik) [reply api/src/main/java/lv/ctco/tpl/bff/graphql/models/JokeModels.java]: thx, fixed

**Comment by trioletas** [comment]
lgtm

**Pull request merged**